### DISCUSSION TITLE
Can't Convert a DraftIssue to An Issue using GraphQL API

### DISCUSSION THREAD

**Discussion by ABIBOO-Team** [community/community#84876] [Apps, API and Webhooks]
### Select Topic Area

Show & Tell

### Body

Hello I hope you're doing great.

I Have been using the GraphQL API for a few moments now in an Integration project, and i'm having a Problem Converting Draft Issues in a Github Project ( Created y using the [addProjectV2DraftIssue](https://docs.github.com/en/graphql#addprojectv2draftissue)  Mutation ) Into an Issue.

The only possible Way that I could see is to Directly Create an Issue ( Using the [createIssue](https://docs.github.com/en/graphql/reference/mutations#createissue) Mutation and Link it to the right project via the parameter **projectIds** ) But when i try to do this i'm having the following error : 

```
{'data': {'createIssue': None},
 'errors': [{'locations': [{'column': 9, 'line': 2}],
             'message': 'Could not resolve to Project node with the global id '
                        "of 'XXXXXXXX'.",
             'path': ['createIssue'],
             'type': 'NOT_FOUND'}]}
```

PS : I already have read-write access to the scopes : **projects**, **issues**. and i can create Issues via the API without linking them to projects.

Anyone has some ideas ?

**Comment by Naimulnobel** [comment]
The error message you're seeing suggests that the GraphQL API can't resolve the project ID you're providing. This could be due to a few reasons:

1. The project ID is incorrect: Make sure you're using the correct project ID. You can fetch the project ID using the `node_id` field from the [List repository projects](https://docs.github.com/en/rest/reference/projects#list-repository-projects) endpoint in the GitHub REST API.

2. The project ID is not a global ID: In GraphQL, IDs are global across the entire GitHub product. This means that you need to use the global ID of the project, not the database ID. You can convert a database ID to a global ID using the `idFromDatabaseId` function in the GraphQL API.

3. The project is not visible to the token: Make sure the token you're using has the correct permissions to access the project. If the project is in a repository, the token needs the `repo` scope. If the project is user- or organization-wide, the token needs the `read:org` scope.

Here's an example of how to create an issue and link it to a project using the GraphQL API:

```graphql
mutation {
  createIssue(input: {repositoryId: "REPO_ID", title: "Issue title", projectIds: ["PROJECT_ID"]}) {
    issue {
      id
      title
      url
    }
  }
}
```

Replace `"REPO_ID"` and `"PROJECT_ID"` with your repository and project global IDs.

**Comment by V5Studio** [comment]
it seems this option in api was added with 3.15
[convertprojectv2draftissueitemtoissue](https://docs.github.com/en/graphql/reference/mutations#convertprojectv2draftissueitemtoissue)
